tests/components/letpot/test_init.py

```diff
@@ -0,0 +1,98 @@
+"""Test the LetPot integration initialization and setup."""
+
+from unittest.mock import MagicMock
+
+from letpot.exceptions import LetPotAuthenticationException, LetPotConnectionException
+import pytest
+
+from homeassistant.components.letpot.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.freeze_time("2025-01-31 00:00:00")
+async def test_load_unload_config_entry(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_client: MagicMock,
+    mock_device_client: MagicMock,
+) -> None:
+    """Test config entry loading/unloading."""
+    await setup_integration(hass, mock_config_entry)
+
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_client.refresh_token.mock_calls) == 0  # Didn't refresh valid token
+    assert len(mock_client.get_devices.mock_calls) == 1
+    assert len(mock_device_client.subscribe.mock_calls) == 1
+    assert len(mock_device_client.get_current_status.mock_calls) == 1
+
+    await hass.config_entries.async_unload(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert not hass.data.get(DOMAIN)
```

Comment:
```suggestion
```

Reply:
Copied this from peblar ;)

How would I test that everything is unloaded? Or can I just believe HA will do it correctly?

Reply:
if the entry state is unloaded that is fine.

This is still a remnant for integrations using hass.data[DOMAIN], which you don't do so this check doesn't make sense